tools/gbench/report.py

```diff
@@ -93,9 +94,111 @@ def filter_benchmark(json_orig, family, replacement=""):
     return filtered
 
 
+def get_unique_benchmark_names(json):
+    """
+    While *keeping* the order, give all the unique 'names' used for benchmarks.
+    """
+    seen = set()
+    uniqued = [x['name'] for x in json['benchmarks']
+               if x['name'] not in seen and
+               (seen.add(x['name']) or True)]
+    return uniqued
+
+
+def get_shared_members(list1, list2):
+    """
+    Given two lists, get a new list consisting of the elements only contained
+    in *both of the input lists*, while preserving the ordering.
+    """
+    seen_in_list1 = set(list1)
```

Comment:
```python
return [x for x in list1 if value in list2]
```

i think will do the same. 

```
>>> l = [3,1,2,4]
>>> m = [3,2,4,5]
>>> 
>>> [x for x in l if x in m]
[3, 2, 4]
>>> [x for x in m if x in l]
[3, 2, 4]
>>> 
```

Reply:
Thanks!
I think i was thinking along the lines that looking up in `set()` would be better,
but i guess we don't really care here.